alica_viewer/include/alica_viewer/alica_plan_parser.h

```diff
@@ -0,0 +1,369 @@
+#ifndef ALICA_VIEWER_ALICA_PLAN_PARSER_H
+#define ALICA_VIEWER_ALICA_PLAN_PARSER_H
+
+#include <SystemConfig.h>
+#include <engine/PlanRepository.h>
+#include <engine/Types.h>
+#include <engine/containers/PlanTreeInfo.h>
+#include <engine/model/EntryPoint.h>
+#include <engine/model/Plan.h>
+#include <engine/model/State.h>
+#include <engine/model/Task.h>
+#include <engine/parser/PlanParser.h>
+#include <supplementary/AgentID.h>
+#include <supplementary/AgentIDManager.h>
+
+namespace alica
+{
+
+class PlanTree;
+class AgentInfo;
+
+typedef std::unordered_map<const supplementary::AgentID*, AgentInfo, supplementary::AgentIDHash, supplementary::AgentIDEqualsComparator> AgentInfoMap;
+typedef std::unordered_map<const supplementary::AgentID*, PlanTree*, supplementary::AgentIDHash, supplementary::AgentIDEqualsComparator> PlanTreeMap;
+typedef std::unordered_map<int64_t, std::vector<PlanTree*>> PlanTreeVectorMap;
```

Comment:
This would be nicer as 
using PlanTreeVectorMap = std::unordered_map<int64_t, std::vector<std::unique_ptr<PlanTree>>>

to mark ownership